### PR TITLE
Move to Julia v1.7

### PR DESCRIPTION
**Pull request by navidcy**

### CODE REVIEW

**Comment by navidcy** [comment]
Precedes #2090

File: src/Operators/spacings_and_areas_and_volumes.jl [hunk 1]
```diff
@@ -102,10 +102,10 @@ end
...
 #####
 
-@inline Δxᶠᵃᵃ(i, j, k, grid::RG)  =  @inbounds grid.Δxᶠᵃᵃ[i]
-@inline Δxᶜᵃᵃ(i, j, k, grid::RG)  =  @inbounds grid.Δxᶜᵃᵃ[i]
-@inline Δyᵃᶠᵃ(i, j, k, grid::RG)  =  @inbounds grid.Δyᵃᶠᵃ[j]
-@inline Δyᵃᶜᵃ(i, j, k, grid::RG)  =  @inbounds grid.Δyᵃᶜᵃ[j]
+@inline Δxᶠᵃᵃ(i, j, k, grid::RG)  =  CUDA.@allowscalar @inbounds grid.Δxᶠᵃᵃ[i]
+@inline Δxᶜᵃᵃ(i, j, k, grid::RG)  =  CUDA.@allowscalar @inbounds grid.Δxᶜᵃᵃ[i]
+@inline Δyᵃᶠᵃ(i, j, k, grid::RG)  =  CUDA.@allowscalar @inbounds grid.Δyᵃᶠᵃ[j]
+@inline Δyᵃᶜᵃ(i, j, k, grid::RG)  =  CUDA.@allowscalar @inbounds grid.Δyᵃᶜᵃ[j]
```
Comment (glwagner) [inline src/Operators/spacings_and_areas_and_volumes.jl:108]: no

Comment (glwagner) [reply src/Operators/spacings_and_areas_and_volumes.jl:108]: If there are scalar operations here, we are ruined and the code will no longer run fast on GPU.

Comment (navidcy) [reply src/Operators/spacings_and_areas_and_volumes.jl:108]: I reverted these!

Comment (glwagner) [reply src/Operators/spacings_and_areas_and_volumes.jl:108]: :-D

File: docs/src/simulation_tips.md [hunk 2]
```diff
@@ -260,22 +259,22 @@ without that step we get an error:
...
 
 
 ```julia
-julia> using CUDA; CUDA.allowscalar(true)
+julia> using CUDA
 
-julia> model.velocities.u.data
-3×3×3 OffsetArray(::CuArray{Float64,3}, 0:2, 0:2, 0:2) with eltype Float64 with indices 0:2×0:2×0:2:
+julia> CUDA.@allowscalar model.velocities.u.data
+3×3×3 OffsetArray(::CuArray{Float64, 3, CUDA.Mem.DeviceBuffer}, 0:2, 0:2, 0:2) with eltype Float64 with indices 0:2×0:2×0:2:
```
Comment (navidcy) [inline docs/src/simulation_tips.md:277]: @tomchor I tried this and it was **not** working on my HPC with Julia v1.7.

Does this works for you on Julia v1.6? Do we need to update anything here?

Comment (tomchor) [reply docs/src/simulation_tips.md:277]: It definitely worked for me when I wrote this with 1.6. I'll test that next time I get access to a GPU (should be this upcoming week), but feel free to replace it for something that works on 1.7 for you!

**Comment by navidcy** [comment]
It'll work!

**Comment by tomchor** [comment]
@navidcy I'm pretty sure https://github.com/CliMA/Oceananigans.jl/pull/2502 fixed at least part of the failing tests here. I'm not sure how to deal with some of the conflicts so I'll leave the merging of main to you (unless you're okay with me taking a crack at it...)

**Comment by navidcy** [comment]
I’ll deal w it in a bit!

**Comment by simone-silvestri** [comment]
merged main, let's see what happens

**Comment by tomchor** [comment]
I don't know how [5c8a763](https://github.com/CliMA/Oceananigans.jl/pull/2307/commits/5c8a763868bfcd244b84ad07122d11647f4f6994) almost passed (everything except for doctests) but then a minor tweak on [4e75ed3](https://github.com/CliMA/Oceananigans.jl/pull/2307/commits/4e75ed3f3222e25d51d24a4c74018e41cbe5d0ba) made a bunch of tests fail...

**Comment by navidcy** [comment]
I stopped buildkite [5c8a763](https://github.com/CliMA/Oceananigans.jl/pull/2307/commits/5c8a763868bfcd244b84ad07122d11647f4f6994) midway so it's not that there was only docs build that were failing....

**Comment by tomchor** [comment]
> I stopped buildkite [5c8a763](https://github.com/CliMA/Oceananigans.jl/pull/2307/commits/5c8a763868bfcd244b84ad07122d11647f4f6994) midway so it's not that there was only docs build that were failing....

Ah, I see. I hadn't noticed that.

**Comment by navidcy** [comment]
@simone-silvestri we may need to start working on this file-by-file...

**Comment by glwagner** [comment]
I'd like to help and hopefully clean up or simplify the tests along the way.

**Comment by navidcy** [comment]
I'll close it as stale. Julia v1.8 is [coming up soon](https://github.com/JuliaLang/julia/releases/tag/v1.8.0) ;)

**Pull request closed without merging**